setupmeta/model.py

```diff
@@ -414,19 +419,46 @@ def checked_file(self, folder, filename):
 class SetupMeta(Settings):
     """ Find usable definitions throughout a project SetupPy SetupMeta """
 
-    def __init__(self, upstream):
+    def __init__(self):
         """
         :param upstream: Either a dict or Distribution
         """
         Settings.__init__(self)
-        self.attrs = MetaDefs.dist_to_dict(upstream)
+        self.attrs = {}
+
+    def preprocess(self, upstream):
+        self.find_project_dir(MetaDefs.dist_to_dict(upstream).pop("_setup_py_path", None))
+
+        for require_field in ('install_requires', 'tests_require'):
+            value = getattr(upstream, require_field)
+            if isinstance(value, str if sys.version_info.major >= 3 else basestring) and value.startswith('@'):  # noqa: E501, F821 (basestring used by Python 2)
+                self.add_definition(require_field, value, EXPLICIT)
+                self.add_definition(require_field, requirements_from_file(value[1:]) or [], source=value[1:], override=True)
```

Comment:
Line 435 shouldn't be needed, or is it just to show the original value in the `explain` output?